test/Libraries/RevitIntegrationTests/WorkflowTests.cs

```diff
@@ -0,0 +1,221 @@
+                var element = GetPreviewValueAtIndex(structuralFraming, i) as StructuralFraming;
+                Assert.IsNotNull(element);
+            }
+
+        }
+
+
+        [Test]
+        [TestModel(@".\Workflow\DynamoRevit\DynamoSample.rvt")]
+        public void Test_AdaptiveComponent1()
+        {
+            string samplePath = Path.Combine(workingDirectory, @".\Workflow\DynamoRevit\02_Adaptive Component Placement 1.dyn");
+            string testPath = Path.GetFullPath(samplePath);
+            ViewModel.OpenCommand.Execute(testPath);
+            RunCurrentModel();
+            var model = ViewModel.Model;
+            Assert.AreEqual(11, model.CurrentWorkspace.Nodes.Count);
+            Assert.AreEqual(10, model.CurrentWorkspace.Connectors.Count());
+
+            //check AdaptiveComponent.ByPoint
+            var AC = "357e7a53-361c-4c1e-81ae-83e16213a39a";
+            AssertPreviewCount(AC, 9);
+
+            //check elements in AdaptiveComponent.ByPoint are not null
+            for (int i = 0; i < 9; i++)
+            {
+                var element = GetPreviewValueAtIndex(AC, i) as AdaptiveComponent;
+                Assert.IsNotNull(element);
+            }
+        }
+
+        [Test]
+        [TestModel(@".\Workflow\DynamoRevit\DynamoSample.rvt")]
+        public void Test_AdaptiveComponent2()
+        {
+            string samplePath = Path.Combine(workingDirectory, @".\Workflow\DynamoRevit\03_Adaptive Component Placement 2.dyn");
+            string testPath = Path.GetFullPath(samplePath);
+            ViewModel.OpenCommand.Execute(testPath);
+            RunCurrentModel();
+            var model = ViewModel.Model;
+            Assert.AreEqual(17, model.CurrentWorkspace.Nodes.Count);
+            Assert.AreEqual(19, model.CurrentWorkspace.Connectors.Count());
+
+            //check AdaptiveComponent.ByPoint
+            var AC = "357e7a53-361c-4c1e-81ae-83e16213a39a";
+            AssertPreviewCount(AC, 9);
+
+            //check elements in AdaptiveComponent.ByPoint are not null
+            for (int i = 0; i < 9; i++)
+            {
+                var element = GetPreviewValueAtIndex(AC, i) as AdaptiveComponent;
+                Assert.IsNotNull(element);
+            }
+
+        }
+
+        [Test]
+        [TestModel(@".\Workflow\DynamoRevit\DynamoSample.rvt")]
+        public void Test_ImportSolid()
+        {
+            string samplePath = Path.Combine(workingDirectory, @".\Workflow\DynamoRevit\04_ImportSolid.dyn");
+            string testPath = Path.GetFullPath(samplePath);
+            ViewModel.OpenCommand.Execute(testPath);
+            RunCurrentModel();
+            var model = ViewModel.Model;
+            Assert.AreEqual(16, model.CurrentWorkspace.Nodes.Count);
+            Assert.AreEqual(16, model.CurrentWorkspace.Connectors.Count());
+
+            //check elements in ImportInstance.ByGeometries are not null
+            Assert.IsNotNull(GetPreviewValue("e3fedc00-247a-4971-901c-7fcb063344c6"));
```

Comment:
it should be validating what should be the type of output. Please add "as Solid" or whatever output your are getting from ImportGeometries node.